subprojects/docs/src/docs/userguide/declaring_dependencies.adoc

```diff
@@ -90,6 +90,142 @@ By default, Gradle caches changing versions of dependencies for 24 hours. Within
 
 Gradle is flexible enough to treat any version as changing version e.g. if you wanted to model snapshot behavior for an Ivy module. All you need to do is to set the property link:{javadocPath}/org/gradle/api/artifacts/ExternalModuleDependency.html#setChanging-boolean-[ExternalModuleDependency.setChanging(boolean)] to `true`.
 
+[[sub:declaring_dependency_rich_version]]
+=== Rich version declaration
+
+Gradle supports a rich model for declaring versions, which allows to combine different level of version information.
+The terms and their meaning are explained below, from the strongest to the weakest:
+
+`strictly`::
+Any version not matched by this version notation will be excluded.
+This is the strongest version declaration.
+It will cause dependency resolution to fail if no version acceptable by this clause can be selected.
+This term supports dynamic versions.
++
+When defined, overrides previous `require` declaration and clears previous `reject`.
+
+`require`::
+Implies that the selected version cannot be lower than what `require` accepts but could be higher through conflict resolution, even if higher has an exclusive higher bound.
+This is what a direct version on a dependency translates to.
+This term supports dynamic versions.
++
+When defined, overrides previous `strictly` declaration and clears previous `reject`.
+
+`prefer`::
+This is a very soft version declaration.
+It applies only if there is no stronger non dynamic opinion on a version for the module.
+This term does not support dynamic versions.
++
+Definition can complement `strictly` or `require`.
+
+There is also an additional term outside of the level hierarchy:
+
+`reject`::
+Declares that specific version(s) are not accepted for the module.
+This will cause dependency resolution to fail if the only versions selectable are also rejected.
+This term supports dynamic versions.
+
+The following table gives a couple use cases and how to combine the different terms for a rich version declaration:
+
+.Rich version use cases
+[%header%autowidth,compact]
+|===
+| Which version(s) of this dependency are acceptable? | `strictly` | `require` | `prefer` | `rejects` | Selection result
+
+| Tested with version `1.5`, believe all future versions should work
+|
+| 1.5
+|
+|
+| Any version starting from `1.5`, equivalent of `org:foo:1.5`. An upgrade to `2.4` is accepted.
+
+| Tested with `1.5`, soft constrain upgrades according to semantic versioning
+|
+| [1.0, 2.0[
+| 1.5
+|
+| Any version between `1.0` and `2.0`, `1.5` if nobody else cares. An upgrade to `2.4` is accepted. +
+🔒
+
+| Tested with `1.5`, restrict upgrades according to semantic versioning
```

Comment:
:o: Maybe somehting like "Tested with 1.5, but follow semantic versioning" would be clearer?